File: src/Entity/Projection.php
<?php

namespace App\Entity;

use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;

/**
 * Projection
 *
 * @ORM\Table(name="projection")
 * @ORM\Entity

 */
class Projection
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     * @Groups("post:read")
     */
    private $id;

    /**
     * @var string
     *
     * @ORM\Column(name="nom", type="string", length=255, nullable=false)
     * @Groups("post:read")
     */
    private $nom;

    /**
     * @var string
     *
     * @ORM\Column(name="genre", type="string", length=255, nullable=false)
     * @Groups("post:read")
     */
    private $genre;

    /**
     * @var int
     *
     * @ORM\Column(name="age_recommande", type="integer", nullable=false)
     * @Groups("post:read")
     */
    private $ageRecommande;

    /**
     * @var string
     *
     * @ORM\Column(name="duree", type="string", length=44, nullable=false)
     * @Groups("post:read")
     */
    private $duree;

    /**
     * @var string
     *
     * @ORM\Column(name="image", type="string", length=10000, nullable=false)
     * @Groups("post:read")
     */
    private $image;

    /**
     * @ORM\OneToMany(targetEntity=ProgrammerFilm::class, mappedBy="projection",cascade={"all"},orphanRemoval=true)

     */
    private $programmerFilms;

    public function __construct()
    {
        $this->programmerFilms = new ArrayCollection();
    }
    /**
     * @var string
     *
     * @ORM\Column(name="description", type="string", length=44, nullable=false)
     * @Groups("post:read")
     */
    private $description;


    /**
     * @return int
     */
    public function getId(): int
    {
        return $this->id;
    }

    /**
     * @param int $id
     */
    public function setId(int $id): void
    {
        $this->id = $id;
    }

    /**
     * @return string
     */
    public function getNom()
    {
        return $this->nom;
    }

    /**
     * @param string $nom
     */
    public function setNom(string $nom): void
    {
        $this->nom = $nom;
    }

    /**
     * @return string
     */
    public function getGenre()
    {
        return $this->genre;
    }

    /**
     * @param string $genre
     */
    public function setGenre(string $genre): void
    {
        $this->genre = $genre;
    }

    /**
     * @return int
     */
    public function getAgeRecommande()
    {
        return $this->ageRecommande;
    }

    /**
     * @param int $ageRecommande
     */
    public function setAgeRecommande(int $ageRecommande): void
    {
        $this->ageRecommande = $ageRecommande;
    }

    /**
     * @return string
     */
    public function getDuree()
    {
        return $this->duree;
    }

    /**
     * @param string $duree
     */
    public function setDuree(string $duree): void
    {
        $this->duree = $duree;
    }

    /**
     * @return string
     */
    public function getImage()
    {
        return $this->image;
    }

    /**
     * @param string $image
     */
    public function setImage($image)
    {
        $this->image = $image;
    }

    /**
     * @return Collection|ProgrammerFilm[]
     */
    public function getProgrammerFilms(): Collection
    {
        return $this->programmerFilms;
    }

    public function addProgrammerFilm(ProgrammerFilm $programmerFilm): self
    {
        if (!$this->programmerFilms->contains($programmerFilm)) {
            $this->programmerFilms[] = $programmerFilm;
            $programmerFilm->setProjection($this);
        }

        return $this;
    }

    public function removeProgrammerFilm(ProgrammerFilm $programmerFilm): self
    {
        if ($this->programmerFilms->removeElement($programmerFilm)) {
            // set the owning side to null (unless already changed)
            if ($programmerFilm->getProjection() === $this) {
                $programmerFilm->setProjection(null);
            }
        }

        return $this;
    }

    /**
     * @return string
     */
    public function getDescription()
    {
        return $this->description;
    }

    /**
     * @param string $description
     */
    public function setDescription(string $description): void
    {
        $this->description = $description;
    }






}

File: src/com/mycompany/myapp/entities/Projection.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.myapp.entities;

import com.codename1.ui.Button;
import javafx.scene.image.ImageView;

/**
 *
 * @author FK Info
 */
public class Projection {
    int id;
    String nom;
    String genre;
    int age_recommande;
    String duree;
   String image;
   String description;
  

  public Projection(int id, String nom, String genre, int age_recommande, String duree, String image, String description) {
        this.id = id;
        this.nom = nom;
        this.genre = genre;
        this.age_recommande = age_recommande;
        this.duree = duree;
        this.image = image;
        this.description = description;
    }
   
    public Projection() {
    }
    public String getImage(){
        return image;
    }
    public void set_Image(String image)
    {
        this.image=image;
    }
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getGenre() {
        return genre;
    }

    public void setGenre(String genre) {
        this.genre = genre;
    }

    public int getAge_recommande() {
        return age_recommande;
    }

    public void setAge_recommande(int age_recommande) {
        this.age_recommande = age_recommande;
    }

    public String getDuree() {
        return duree;
    }

    public void setDuree(String duree) {
        this.duree = duree;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

   @Override
    public String toString() {
        return "projection{" + "id=" + id + ", nom=" + nom + ", genre=" + genre + ", age=" +age_recommande +", duree=" +duree +", img=" +image +", description=" +description + '}';
    }
    
}
